src/Microsoft.ML.Core/Data/UInt128.cs

```diff
@@ -167,7 +167,7 @@ private static void FinalMix(ref ulong h1, ref ulong h2, int len)
         /// that were all zeros, except for the last bit which is one.
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public UInt128 Fork()
+        public RowId Fork()
```

Comment:
>Fork [](start = 21, length = 4)

can be internal, right?

Reply:
I will be moving this and any other types that IDataView needs out into a separate assembly, which will not have InternalsVisibleTo on it. So let's keep whatever things we need as `public`.

Alternatively, the `Lo` and `Hi` fields are public, so we could make these methods static extension methods that were only in ML.NET. (And those extension methods would be internal.)

Reply:
If you look on usage, only thing which we use outside is Combine method.
Part of me wants just delete all this unused code, but since it was written by someone, maybe it can be useful someday.
So I would prefer just hide it behind internal/private.

Just in case I'm talking about Hashing region.


---
In reply to: [240724280](https://github.com/dotnet/machinelearning/pull/1858#discussion_r240724280) [](ancestors = 240724280)

Reply:
Please **do not remove this code**. It is needed for definite purposes as explained in our code documentation.